docs/DeveloperGuide.md

```diff
@@ -4,22 +4,194 @@
+2. The `Ui` class depends on `Scanner` to get inputs from users.
+3. The `Main` class uses `Ui#getNextLine()` to receive `commands` from users.
+4. And `FlashcardFactory` and `GroupFactory` classes uses methods in `Ui` class to complete 
+the execution of various commands.
+
+Given below is the Sequence Diagram for interactions between the `Ui` and `FlashcardFactory` components to complete the
+execution of `deleteCommand`.
+
+![UiSequenceDiagram](images/UiSequenceDiagram.png)
+
+#### Parser Component
+
+The structure of the Parser Component is illustrated in the following diagram:
+
+![ParserComponentDiagram](images/ParserComponent.png)
+
+API : `Parser.java`
+
+1. The `Main` class (not shown in the diagram above) uses the `Parser` class to parse the user command.
+2. The `Parser` class depends on multiple classes including the `FlashcardFactory`, `FlashcardList`, `GroupFactory
+`, `GroupList`, `StudyPlanList` and `Ui` classes to create a `Command` object from user input.
+3. The resulting `Command` object which then can be executed by the `Main` class.
+
+Given below is the Sequence Diagram for interactions between the `Parser` and `Command` components for the
+`parseCommand` API call from `Main` class. We use the interaction between the `Parser` and `DeleteCommand` class as
+ an example: 
+
+![ParserAndCommandSequenceDiagram](images/ParserAndCommandSequenceDiagram.png)
+
+#### Storage Component
+
+{Description for Storage Component}
+
+### Implementation
+
+#### Set Priority Feature
+
+The Set Priority feature allows users to mark a flashcard with a specified priority level. The priority level is
+then reflected when the user requests a list of existing flashcards.
+
+##### Proposed Implementation
+
+`PriorityLevel` is stored as an enum with four fields: LOW, MEDIUM, HIGH and DEFAULT.
+
+The feature implements the following operations:
+
+- `Flashcard#setPriorityLevel()` - Sets a flashcard to a specified priority level (LOW/MEDIUM/HIGH).
+- `Flashcard#getPriorityAsString()` - Return an icon indicating the flashcard's priority level.
+
+The following class diagram shows the structures relevant to the Set Priority feature:
+
+![SetPriorityFeatureClassDiagram](images/SetPriorityFeatureClassDiagram.png)
+
+Given below is an example usage scenario and how the set priority mechanism behaves at each step.
+
+Step 1. The user launches the application and creates a new `Flashcard` (of type Event, Person, or Other), 
+with a default priority level of DEFAULT.
+
+Step 2. The user executes `priority [flashcard index] [priority level]` command and the `setPriorityLevel` 
+operation is invoked. Subsequently, the flashcard of specified index is retrieved from the main `FlashcardList` 
+instance. Its priority level is set to the specified priority.
+
+The following sequence diagram shows the relevant interactions behind `setPriorityLevel`:
+
+![setPriorityLevelSequenceDiagram](images/setPriorityLevelSequenceDiagram.png)
```

Comment:
This sequence diagram is well illustrated and easy to understand as well. Maybe the activity bars for an object that is returning a value that has been returned to it could be spaced out a bit (see PriorityCommand returning to Command which returns to Main). However there is nothing wrong with how it is currently drawn.